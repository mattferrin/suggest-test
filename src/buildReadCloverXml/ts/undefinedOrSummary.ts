import { ReadCloverXmlOutput } from "./buildReadCloverXml";

export interface SummaryItem {
  readonly conditionals: number | null | undefined;
  readonly path: string | null | undefined;
  readonly statements: number | null | undefined;
  readonly uncoveredConditionals: number;
  readonly uncoveredStatements: number;
  readonly score: number;
  readonly hasUnmarkedConditional: boolean;
}

export function undefinedOrSummary(
  summary: ReadonlyArray<SummaryItem | undefined> | undefined
): ReadCloverXmlOutput {
  if (summary === undefined || summary === null) {
    return {
      tag: "errors",
      errors: [
        new Error(
          "The parsed clover.xml has an undefined or null property somewhere"
        ),
      ],
    };
  } else {
    return {
      tag: "summary",
      summaryArray: summary,
      summaryString: { tag: "not-ready" },
    };
  }
}
