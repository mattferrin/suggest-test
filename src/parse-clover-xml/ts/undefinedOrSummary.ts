import { ReadCloverXmlOutput } from "./readCloverXml";

export interface SummaryItem {
  readonly path: string | null | undefined;
  readonly uncoveredStatements: number;
  readonly statements: number | null | undefined;
  readonly uncoveredConditionals: number;
  readonly conditionals: number | null | undefined;
}

export function undefinedOrSummary(
  summary: ReadonlyArray<SummaryItem | undefined> | undefined
): ReadCloverXmlOutput {
  if (summary === undefined || summary === null) {
    return {
      tag: "errors",
      errors: [
        new Error(
          "The parsed clover.xml has an undefined or null property somewhere"
        ),
      ],
    };
  } else {
    return { tag: "summary", summary };
  }
}
