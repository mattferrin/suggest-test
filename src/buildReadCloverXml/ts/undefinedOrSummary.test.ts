import { undefinedOrSummary } from "./undefinedOrSummary";

function snap(input: {
  stabileSortId: string;
  comment: string | null;
  summary: Parameters<typeof undefinedOrSummary>[0];
}): void {
  it(input.stabileSortId, () => {
    /** setup mocks */

    /** unit under test */
    const result = undefinedOrSummary(input.summary);

    /** assertions */
    expect({
      [`${input.comment ?? "TODO: comment"}`]: {
        result,
      },
    }).toMatchSnapshot();
  });
}

snap({
  stabileSortId: "4e1c9e6f-a5ef-48c7-abcd-040d76113c79",
  comment: "clover file has undefined or null",
  summary: undefined,
});

snap({
  stabileSortId: "86a79ca4-df17-45fb-8d9f-cba3802eb3b8",
  comment: "summary array defined AND summary string not computed yet",
  summary: "summary arg" as any,
});
