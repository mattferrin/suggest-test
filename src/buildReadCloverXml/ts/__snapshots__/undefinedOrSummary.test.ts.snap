// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`4e1c9e6f-a5ef-48c7-abcd-040d76113c79 1`] = `
{
  "clover file has undefined or null": {
    "result": {
      "errors": [
        [Error: The parsed clover.xml has an undefined or null property somewhere],
      ],
      "tag": "errors",
    },
  },
}
`;

exports[`86a79ca4-df17-45fb-8d9f-cba3802eb3b8 1`] = `
{
  "summary array defined AND summary string not computed yet": {
    "result": {
      "summaryArray": "summary arg",
      "summaryString": {
        "tag": "not-ready",
      },
      "tag": "summary",
    },
  },
}
`;
